quil-rs/src/program/graph.rs

```diff
@@ -691,3 +691,154 @@ impl<'a> ScheduledProgram<'a> {
         label
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::instruction::Pragma;
+    use crate::program::{MatchedFrames, MemoryAccesses};
+    use insta::assert_debug_snapshot;
+
+    mod custom_handler {
+        use super::*;
+        use crate::instruction::PragmaArgument;
+        use crate::program::frame::FrameMatchCondition;
+
+        fn get_custom_handler() -> InstructionHandler {
```

Comment:
This handler looks like it tests all the right stuff.

docstring with intent would be useful, though